### PR TITLE
make pretrained backbone flexible in predictor

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- solves : #991 

Any feedback is welcome :hugs:

### CODE REVIEW

File: doctr/models/detection/zoo.py [hunk 1]
```diff
@@ -46,14 +46,20 @@ def _predictor(arch: Any, pretrained: bool, assume_straight_pages: bool = True,
...
+            assume_straight_pages=assume_straight_pages,
+        )
     else:
         if not isinstance(arch, (detection.DBNet, detection.LinkNet)):
             raise ValueError(f"unknown architecture: {type(arch)}")
 
         _model = arch
         _model.assume_straight_pages = assume_straight_pages
 
+    kwargs.pop("pretrained_backbone", None)
```
Comment (frgfm) [inline doctr/models/detection/zoo.py:61]: no need for the second arg since we don't capture the output

**Review by frgfm — APPROVED** [review]
Thanks Felix! A slight comment but that's quite alright

File: doctr/models/detection/zoo.py (hunk 1 above)
Comment (felixdittrich92) [reply doctr/models/detection/zoo.py:61]: mh .. i have tried this but without success.
However, I don't remember what exactly the error was :sweat_smile:

**Comment by felixdittrich92** [comment]
will be merged after #1063 to avoid failing CI

File: doctr/models/detection/zoo.py (hunk 1 above)
Comment (frgfm) [reply doctr/models/detection/zoo.py:61]: oh perhaps it fails if the key doesn't exist without the default value

**Pull request merged**